src/cpp/core/system/PosixSystemTests.cpp

```diff
@@ -43,10 +45,11 @@ static std::string getNoGroupName()
    else if (getgrnam("nogroup"))
       group = "nogroup"; // Debian/Ubuntu
 
-   expect_false(group.empty());
```

Comment:
Not sure if this expectation should be part of the helper function, or part of the test case itself?

Reply:
Was it causing a build warning?

I put this here to make it easier to spot any future quirks where a container doesn't have one of the expected groups.

Reply:
Ah I see on Mac it flags getNoGroupName as unused; go ahead and wrap that function in `#ifdef __linux__`.

Reply:
Sounds good; I'll leave it in.